### ISSUE TITLE
KT-23526 - IntelliJ debugger ignores breakpoints in Kotlin scripts

### ISSUE THREAD

**Issue by bamboo** [gradle/kotlin-dsl-samples#1040]
[KT-23526](https://youtrack.jetbrains.com/issue/KT-23526)

The debugger doesn't ignore breakpoints in class and object members so a workaround is to declare an `object` in the script and put the breakpoint there:

![precompiled-script-plugin-breakpoints.png](https://images.zenhubusercontent.com/5452207c7c6ccc6277a5cc26/17e09be7-300f-4d3e-8819-8d10257ce9d3)

Top-level breakpoint in line 10 above is ignored but the breakpoint in line 6 is not because it's set within an `object` member.

**Comment by fuzzyweapon** [comment]
This probably falls under the same category as unsupported top level breakpoints or other reported areas (but I'm mentioning it anyways just in case it doesn't).  On Kotlin DSL shipped with 5.0, this affects top level functions in the script.

As you can see below, the callsite of this function and underlying project structure is less that straightforward, so I've included these for posterity's sake.  As I don't have the time to narrow this down, I'll watch this issue and test this case once the linked issues are merged into a stable release.

```
// shared.build.gradle.kts applied to the root project of a multiproject included build (we'll call this rootIMP) via `apply(from = "<absolute_path_outside_of_project_structure>")`

// Breakpoint indicated at *

fun rootIMPFoo(project: Project) {
    with(project) {
        afterEvaluate {
*           println("Breakpoint here!"
        }
    }
}

// Including this for posterity's sake (for my own regression) in case it's a part of the root cause.
val rootIMPFooCaller(p: Project) by extra {
    // Just a helper that makes the lambda always return Unit
    toUnit { 
        with(p) {
            rootIMPFoo(p)
        }
    }
}
```

```
// rootIMP's build.gradle.kts

apply(from = "<absolute_path_to>/shared.gradle.kts")

// ...redacted configuration

val rootIMPFoo: (Project) -> Unit by extra

subprojects {
    val subProject = this
    plugins.withType<KotlinPluginWrapper> {
        rootIMPFoo(subProject)
    }
}
```

**Comment by JLLeitschuh** [comment]
There are also issues with breakpoints in plugin libraries. I've only been able to get breakpoints in plugins to work if I also declare the plugins as runtime dependencies in some sub-project like the `buildsrc` or in some project's `dependencies` block.

**Comment by eskatos** [comment]
Fixed in Kotlin 1.3.50 🎉
